### PR TITLE
refactor(modal): remove CSS-in-JS [skip ci]

### PR DESCRIPTION
**Pull request by colbr**
- Removes old CSS-in-JS styling from the `Modal` component.
- Adds `Modal.module.scss` for styling

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -3653,6 +3653,11 @@
     "@testing-library/dom" "^8.0.0"
     "@types/react-dom" "<18.0.0"
 
+"@testing-library/user-event@^14.4.3":
+  version "14.4.3"
+  resolved "https://registry.yarnpkg.com/@testing-library/user-event/-/user-event-14.4.3.tgz#af975e367743fa91989cd666666aec31a8f50591"
+  integrity sha512-kCUc5MEwaEMakkO5x7aoD+DLi02ehmEM2QCGWvNqAS1dV/fAvORWEjnjsEIvml59M7Y5kCkWN6fCCyPOe8OL6Q==
+
```
Comment (joonchoi0918) [inline yarn.lock:3660]: You got this issue. The another developer didn't update it in his PR 😄

File: src/components/Modal/Modal.module.scss [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    background: rgba(0, 0, 0, 0.9);
+}
+
+.Content {
+    min-width: 300px;
+    padding: 30px;
+    border-radius: 16px;
+    border: 1px solid #3b3b3b;
+
+    background: #0a0a0a;
```
Comment (joonchoi0918) [inline src/components/Modal/Modal.module.scss:22]: Don't we use the colors defined in `styles/_colors.scss`?

File: src/components/Modal/Modal.module.scss [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    display: grid;
+    position: fixed;
+    top: 0;
+    left: 0;
+    right: 0;
+    bottom: 0;
+
+    place-items: center;
+    overflow-y: auto;
+    z-index: 10000;
```
Comment (joonchoi0918) [inline src/components/Modal/Modal.module.scss:11]: I recommend to use some variable for z-index something like `$z-index-100~10000` or  `$z-index-xxs~xxl`

File: src/components/Modal/Modal.module.scss [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+    background: rgba(0, 0, 0, 0.9);
+}
+
+.Content {
+    min-width: 300px;
+    padding: 30px;
+    border-radius: 16px;
```
Comment (joonchoi0918) [inline src/components/Modal/Modal.module.scss:19]: nit:  I think we moved to using `px` from `rem`

File: src/components/Modal/Modal.tsx [hunk 5]
```diff
@@ -8,32 +8,11 @@ import {
...
-  zIndex: 10000
-});
-
-const Content = styled(DialogContent, {
-  minWidth: 300,
-  background: '#0a0a0a',
-  padding: 30,
-  borderRadius: 16,
-  border: '1px solid #3b3b3b'
-});
```
Comment (joonchoi0918) [inline src/components/Modal/Modal.tsx:36]: I really couldn't understand this styled component, but I am happy to see removing this 😄

**Review by joonchoi0918 — APPROVED** [review]
Great Work! I left a few comments about variables and colors. But I know we will get rid of these legacy colors when we use `ThemeColors` and I think it is fine for now. Not sure you used the color which is in the color pallets.

File: src/components/Modal/Modal.tsx (hunk 5 above)
Comment (colbr) [reply src/components/Modal/Modal.tsx:36]: I was just testing out Radix when I made this component - they used Stitches in the example which I more or less copied, haha. I'm glad to purge it too!

File: yarn.lock (hunk 1 above)
Comment (colbr) [reply yarn.lock:3660]: Probably me, haha 🤡 All good, should be sorted once this is merged.

File: src/components/Modal/Modal.module.scss (hunk 2 above)
Comment (colbr) [reply src/components/Modal/Modal.module.scss:22]: Good point, let me fix that

File: src/components/Modal/Modal.module.scss (hunk 3 above)
Comment (colbr) [reply src/components/Modal/Modal.module.scss:11]: I think in this case we wouldn't want anything to share the same zIndex as the `Modal` component, as it's meant to be on top. Let me know if this assumption is incorrect though!

File: src/components/Modal/Modal.module.scss (hunk 4 above)
Comment (colbr) [reply src/components/Modal/Modal.module.scss:19]: Good call - I'll switch them.

File: src/components/Modal/Modal.module.scss (hunk 2 above)
Comment (colbr) [reply src/components/Modal/Modal.module.scss]: Fixed!

File: src/components/Modal/Modal.module.scss (hunk 4 above)
Comment (colbr) [reply src/components/Modal/Modal.module.scss]: Fixed!

**Pull request merged**